src/Camelot.Services.AllPlatforms/ShellIconsCacheService.cs

```diff
@@ -0,0 +1,180 @@
+using System;
+using System.Collections.Generic;
+using Camelot.Services.Abstractions;
+using System.IO;
+using Camelot.Services.Abstractions.Models;
+using Camelot.Images;
+using Avalonia.Media.Imaging;
+using Camelot.Services.Environment.Interfaces;
+using Camelot.Services.Environment.Enums;
+
+namespace Camelot.Services.AllPlatforms;
+
+public class ShellIconsCacheService : IShellIconsCacheService
+{
+    private readonly IShellLinksService _shellLinksService;
+    private readonly IShellIconsService _shellIconsService;
+    private readonly Dictionary<string, Bitmap> _cache = new();
```

Comment:
What happens if the cache grows too large? Should we enforce some size limits?

Reply:
Good question. Cache is limited to number of extensions and apps.
I guess for normal user, if he has 30 extensions and some 100 apps, then it will not grow more than 150-200 icons.
But its a good idea to add some limit.